server/serverimpl.go

```diff
@@ -243,13 +259,40 @@ func (s *server) handleWSConnection(reqCtx context.Context, wsConn *websocket.Co
 
 	sentCustomCapabilities := false
 
+	type webSocketMessage struct {
+		mt       int
+		msgBytes []byte
+		err      error
+	}
+	msgCh := make(chan webSocketMessage)
+
 	// Loop until fail to read from the WebSocket connection.
 	for {
-		msgContext := context.Background()
+		msgContext := s.serverCtx
 		request := protobufs.AgentToServer{}
 
-		// Block until the next message can be read.
-		mt, msgBytes, err := wsConn.ReadMessage()
+		go func() {
```

Comment:
Creating a goroutine for each message seems excessive, this may have a significant cost for tiny messages (benchmark?). Alternatively, can we create one goroutine per connection and keep it running until the connection is closed?

Reply:
fixed 
https://github.com/open-telemetry/opamp-go/pull/380/commits/1e9f46a5a12808489b41f66dffcc97b8edc80e3f
https://github.com/open-telemetry/opamp-go/pull/380/commits/787910f6d3bf6a5ce56ad9032ed0899ba0ce5f10